backend/src/backend/user_session/routers/grid/router.py

```diff
@@ -0,0 +1,496 @@
+# type: ignore
+# for now
+from functools import cache
+from typing import List, Tuple
+import logging
+import os, psutil
+
+import numpy as np
+import orjson
+import xtgeo
+from vtkmodules.util.numpy_support import vtk_to_numpy
+from fastapi import APIRouter, Depends, Request, Response
+from src.backend.auth.auth_helper import AuthenticatedUser, AuthHelper
+from src.backend.primary.routers.grid.schemas import (
+    B64EncodedNumpyArray,
+    GridGeometry,
+    GridIntersection,
+)
+from src.services.sumo_access.grid_access import GridAccess
+from src.services.utils.b64 import b64_encode_numpy
+from src.services.utils.vtk_utils import (
+    VtkGridSurface,
+    get_scalar_values,
+    get_surface,
+    cut_along_polyline,
+    flatten_sliced_grid,
+    xtgeo_grid_to_vtk_explicit_structured_grid,
+    create_polyline,
+    grid_to_numpy,
+    get_triangles,
+)
+from src.services.utils.mpl_utils import visualize_triangles_with_scalars
+from src.services.utils.perf_timer import PerfTimer
+
+router = APIRouter()
+LOGGER = logging.getLogger(__name__)
+process = psutil.Process()
+
+
+@router.get(
+    "/grid_geometry", response_model=GridGeometry
+)  # stating response_model here instead of return type apparently disables pydantic validation of the response (https://stackoverflow.com/a/65715205)
+# type: ignore
+async def grid(
+    request: Request,
+    authenticated_user: AuthenticatedUser = Depends(AuthHelper.get_authenticated_user),
+):
+    """ """
+    case_uuid = request.query_params.get("case_uuid")
+    ensemble_name = request.query_params.get("ensemble_name")
+    grid_name = request.query_params.get("grid_name")
+    realization = request.query_params.get("realization")
+
+    print("Sending data to primary", flush=True)
+    grid_geometry = get_grid_geometry(
+        authenticated_user=authenticated_user,
+        case_uuid=case_uuid,
+        ensemble_name=ensemble_name,
+        grid_name=grid_name,
+        realization=realization,
+    )
+    grid_geometrics = grid_geometry.get_geometrics(allcells=True, return_dict=True)
+    grid_polydata = get_grid_polydata(grid_geometry=grid_geometry)
+
+    points_np = vtk_to_numpy(grid_polydata.polydata.GetPoints().GetData()).ravel().astype(np.float32)
+    polys_np = vtk_to_numpy(grid_polydata.polydata.GetPolys().GetData()).astype(np.int64)
+    points_np = np.around(points_np, decimals=2)
+    # grid_geometry = GridGeometry(points=points_np.tolist(), polys=polys_np.tolist(), **grid_geometrics)
+
+    grid_geometry = GridGeometry(
+        points=b64_encode_numpy(points_np),
+        polys=b64_encode_numpy(polys_np),
+        **grid_geometrics,
+    )
+    return Response(orjson.dumps(grid_geometry.__dict__), media_type="application/json")
```

Comment:
Simpler to use the `from fastapi.responses import ORJSONResponse`? Maybe
```suggestion
    return ORJSONResponse(grid_geometry.dict())
```
works (https://docs.pydantic.dev/latest/usage/exporting_models/#modeldict)